tuna/import_benchmark.py

```diff
@@ -0,0 +1,121 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+""" Module adding frameworks/models/benchmarks"""
+from sqlalchemy.exc import IntegrityError
+
+from tuna.dbBase.sql_alchemy import DbSession
+from tuna.parse_args import TunaArgs, setup_arg_parser
+from tuna.utils.logger import setup_logger
+from tuna.miopen.benchmark import Framework, Model, FrameworkEnum, ModelEnum
+
+LOGGER = setup_logger('import_benchmarks')
+
+
+def parse_args():
+  """Parsing arguments"""
+  parser = setup_arg_parser('Import benchmark performance related items',
+                            [TunaArgs.VERSION])
+  parser.add_argument('--update_framework',
+                      action="store_true",
+                      dest='update_framework',
+                      help='Populate framework table with all framework enums')
+  parser.add_argument('--add_model',
+                      dest='add_model',
+                      type=ModelEnum,
+                      choices=ModelEnum,
+                      help='Populate table with new model and version')
+  parser.add_argument('--print_models',
+                      dest='print_models',
+                      action='store_true',
+                      help='Print models from table')
+  parser.add_argument('--version',
+                      dest='version',
+                      type=str,
+                      default=None,
+                      required=False,
+                      help='Specify model version')
+
+  args = parser.parse_args()
+  if args.add_model and not args.version:
+    parser.error('Version needs to be specified with model')
+  return args
+
+
+def print_models():
+  """Display models from the db table"""
+  with DbSession() as session:
+    models = session.query(Model).all()
+    for model in models:
+      LOGGER.info('model %s version %s ', model.model, model.version)
+  return True
+
+
+def add_model(args):
+  """Add new model and version to the db table"""
+  with DbSession() as session:
+    new_model = Model(model=args.add_model, version=args.version)
+    try:
+      session.add(new_model)
+      session.commit()
+      LOGGER.info('Added model %s with version %s ', args.add_model,
+                  str(args.version))
+    except IntegrityError as err:
+      LOGGER.error(err)
+      return False
+
+  return True
+
+
+def update_frameworks():
+  """Bring DB table up to speed with enums defined in FrameworkEnum"""
+  with DbSession() as session:
+    for elem in FrameworkEnum:
+      try:
+        new_fmk = Framework(framework=elem)
+        session.add(new_fmk)
+        session.commit()
+        LOGGER.info('Added new framework %s', elem)
+      except IntegrityError as ierr:
+        LOGGER.warning(ierr)
+        continue
```

Comment:
Should the order of the frameworks in the table be the same as that of the enum?

Reply:
Yes, i think the above iteration guarantees that